azurelinuxagent/common/protocol/wire.py

```diff
@@ -1092,29 +1089,53 @@ def send_event(self, provider_id, event_str):
                 "Failed to send events:{0}".format(resp.status))
 
     def report_event(self, event_list):
+        max_send_errors_to_report = 5
         buf = {}
+        events_per_request = 0
+        unicode_error_count, unicode_errors = 0, []
+        event_report_error_count, event_report_errors = 0, []
+
         # Group events by providerId
         for event in event_list.events:
-            if event.providerId not in buf:
-                buf[event.providerId] = ""
-            event_str = event_to_v1(event)
-            if len(event_str) >= MAX_EVENT_BUFFER_SIZE:
-                details_of_event = [ustr(x.name) + ":" + ustr(x.value) for x in event.parameters if x.name in
-                                    [GuestAgentExtensionEventsSchema.Name, GuestAgentExtensionEventsSchema.Version,
-                                     GuestAgentExtensionEventsSchema.Operation,
-                                     GuestAgentExtensionEventsSchema.OperationSuccess]]
-                logger.periodic_warn(logger.EVERY_HALF_HOUR,
-                                     "Single event too large: {0}, with the length: {1} more than the limit({2})"
-                                     .format(str(details_of_event), len(event_str), MAX_EVENT_BUFFER_SIZE))
-                continue
-            if len(buf[event.providerId] + event_str) >= MAX_EVENT_BUFFER_SIZE:
-                self.send_event(event.providerId, buf[event.providerId])
-                buf[event.providerId] = ""
-            buf[event.providerId] = buf[event.providerId] + event_str
+            try:
+                if event.providerId not in buf:
+                    buf[event.providerId] = b''
```

Comment:
So to avoid double encoding and also to ensure that the `len(encoded_string) < MAX_EVENT_BUFFER_SIZE`, I changed the `buf[event.providerId]` to save the encoded string. Encode() in py3 converts the string to bytes and keeps it as str in py2. This is the reason I changed it to bytes.